builtin/providers/vsphere/resource_vsphere_virtual_disk.go

```diff
@@ -0,0 +1,250 @@
+package vsphere
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"golang.org/x/net/context"
+)
+
+type virtualDisk struct {
+	size        int
+	vmdkPath    string
+	initType    string
+	adapterType string
+	datacenter  string
+	datastore   string
+}
+
+// Define VirtualDisk args
+func resourceVSphereVirtualDisk() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVSphereVirtualDiskCreate,
+		Read:   resourceVSphereVirtualDiskRead,
+		Delete: resourceVSphereVirtualDiskDelete,
+
+		Schema: map[string]*schema.Schema{
+			// Size in GB
+			"size": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true, //TODO Can this be optional (resize)?
+			},
+
+			"vmdk_path": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true, //TODO Can this be optional (move)?
+			},
+
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "eagerZeroedThick",
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if value != "thin" && value != "eagerZeroedThick" {
+						errors = append(errors, fmt.Errorf(
+							"only 'thin' and 'eagerZeroedThick' are supported values for 'type'"))
+					}
+					return
+				},
+			},
+
+			"adapter_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "ide",
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if value != "ide" && value != "busLogic" && value != "lsiLogic" {
+						errors = append(errors, fmt.Errorf(
+							"only 'ide', 'busLogic', and 'lsiLogic' are supported values for 'adapter_type'"))
+					}
+					return
+				},
+			},
+
+			"datacenter": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"datastore": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceVSphereVirtualDiskCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Creating Virtual Disk")
+	client := meta.(*govmomi.Client)
+
+	vDisk := virtualDisk{
+		size: d.Get("size").(int),
+	}
+
+	if v, ok := d.GetOk("vmdk_path"); ok {
+		vDisk.vmdkPath = v.(string)
```

Comment:
@chrislovecnm the setting of the state usually happens I'm the read func so we are sure that we expected to happen has happened.